src/libstd/sys/unix/thread.rs

```diff
@@ -180,7 +182,7 @@ pub mod guard {
 }
 
 
-#[cfg(any(target_os = "linux",
+#[cfg(any(all(target_os = "linux", not(target_env = "musl")),
```

Comment:
Unfortunately I'm without the ability to build musl-rust right now, so I'm forced to ask rather than just trying - what about this won't work on musl?